dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.ts

```diff
@@ -145,42 +145,43 @@ export class TfmFacilitiesRepo {
    * @param status - The amendment status
    * @returns The found amendments
    */
-  public static async findAmendmentsByFacilityIdAndStatus(facilityId: string | ObjectId, status: AmendmentStatus): Promise<TfmFacilityAmendment[]> {
+  public static async findAmendmentsByFacilityIdAndStatus(facilityId: string | ObjectId, status: AmendmentStatus): Promise<FacilityAmendment[]> {
     const collection = await this.getCollection();
     return await collection
       .aggregate(aggregatePipelines.amendmentsByFacilityIdAndStatus(facilityId, status))
-      .map<TfmFacilityAmendment>((doc) => doc.amendments as TfmFacilityAmendment)
+      .map<FacilityAmendment>((doc) => doc.amendments as FacilityAmendment)
       .toArray();
   }
 
   /**
-   * Finds amendments by the facility id and amendment id
-   * @param facilityId - The facility id
-   * @param amendmentId - The amendment id
-   * @returns The found amendments
-   */
-  public static async findAmendmentsByFacilityIdAndAmendmentId(facilityId: string | ObjectId, amendmentId: string | ObjectId): Promise<Document[]> {
-    const collection = await this.getCollection();
-    return await collection
-      .aggregate(aggregatePipelines.amendmentsByFacilityIdAndAmendmentId(facilityId, amendmentId))
-      .map<Document>((doc) => doc.amendments as Document)
-      .toArray();
-  }
-
-  /**
-   * Finds the amendment with the supplied facility id and amendment id
+   * Finds amendment by facility id and amendment id
    * @param facilityId - The facility id
    * @param amendmentId - The amendment id
    * @returns The found amendment
    */
-  public static async findAmendmentByFacilityIdAndAmendmentId(facilityId: string | ObjectId, amendmentId: string | ObjectId): Promise<Document | null> {
+  public static async findOneAmendmentByFacilityIdAndAmendmentId(
+    facilityId: string | ObjectId,
+    amendmentId: string | ObjectId,
+  ): Promise<(FacilityAmendment & { ukefFacilityId: string | null }) | undefined> {
     const collection = await this.getCollection();
-    const results = await collection
-      .aggregate(aggregatePipelines.amendmentsByFacilityIdAndAmendmentId(facilityId, amendmentId))
```

Comment:
I think its more clear to use `findOne` - since there should only be one matching facility. This also means the types are inferred automatically